src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -390,6 +387,19 @@ MetadataCreateCommands(void)
 		ObjectAddressSet(tableAddress, RelationRelationId, relationId);
 		EnsureDependenciesExistOnAllNodes(&tableAddress);
 
+		/*
+		 * Ensure sequence dependencies and mark them as distributed
+		 */
+		List *attnumList = NIL;
+		List *dependentSequenceList = NIL;
+		GetDependentSequencesWithRelation(relationId, &attnumList, &dependentSequenceList,
+										  0);
+		Oid sequenceOid = InvalidOid;
+		foreach_oid(sequenceOid, dependentSequenceList)
+		{
+			EnsureSequenceDependenciesAndMarkDist(sequenceOid);
+		}
```

Comment:
Makes sense, but this code could use a bit of a (separate) refactor to take the dependency creation logic out of MetadataCreateCommands.

Reply:
It's not the best place, but there is no obvious better place for the dependency creation logic. Here it goes well with the dependency creation logic for the table. Maybe as a future note we can move both of them if possible.